frontend/src/libs/components/modal/steps/delete-account/delete-acount-form.tsx

```diff
@@ -0,0 +1,104 @@
+import { type ChangeEvent, type FC } from 'react';
+
+import { Button, Checkbox, Input } from '#libs/components/components.js';
+import { useAppForm, useCallback, useState } from '#libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  onNext?: () => void;
+  onClose?: () => void;
+};
+
+const DeleteAccountForm: FC<Properties> = ({ onNext, onClose }) => {
+  const { control, errors } = useAppForm({
+    defaultValues: {
+      describeYourSituation: '',
+    },
+    mode: 'onSubmit',
+  });
+
+  const [isChecked, setIsChecked] = useState({
+    'checkbox1': false,
+    'checkbox2': false,
+    'checkbox3': false,
+    'checkbox4': false,
+  });
+
+  const handleCheckboxChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>): void => {
+      const { name, checked } = event.target;
+      setIsChecked((previousState) => {
+        return { ...previousState, [name]: checked };
+      });
+    },
+    [],
+  );
+
+  const isNextDisabled = !Object.values(isChecked).some(Boolean);
+
+  return (
+    <div className={styles['modal-body']}>
+      <form className={styles['form']}>
```

Comment:
all forms should be controlled using useAppForm

Reply:
I implemented the other comments, still working on this one.

Reply:
Done✅